### PR TITLE
Feature/donor view merge into develop

### PR DESCRIPTION
**Pull request by p-maks**
Donor view work

### CODE REVIEW

File: root/api/src/main/java/org/patientview/api/model/Pathway.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ * Pathway represents pathway for the User eg Donorview
+ */
+public class Pathway implements Serializable {
+
+    private Long id;
+    private PathwayTypes pathwayType;
+    private BaseUser user;
+    private BaseUser creator;
+    private Map<String, Stage> stages;
+    private Date lastUpdate;
```
Comment (jamesrossiter) [inline root/api/src/main/java/org/patientview/api/model/Pathway.java]: missing last updated by

File: root/api/src/main/java/org/patientview/api/service/NoteService.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     * Add an Note for a User.
+     *
+     * @param userId ID of User adding Note
+     * @param Note   a Note object to be created
+     * @throws ResourceNotFoundException
+     * @throws ResourceForbiddenException
+     */
+    @RoleOnly(roles = {RoleName.UNIT_ADMIN, RoleName.STAFF_ADMIN,
+            RoleName.DISEASE_GROUP_ADMIN, RoleName.SPECIALTY_ADMIN, RoleName.GLOBAL_ADMIN,
+            RoleName.GP_ADMIN, RoleName.UNIT_ADMIN})
```
Comment (jamesrossiter) [inline root/api/src/main/java/org/patientview/api/service/NoteService.java]: unit admin in twice

File: root/api/src/main/java/org/patientview/api/service/impl/PathwayServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        if (entity == null) {
+            throw new ResourceNotFoundException("Could not find pathway");
+        }
+
+        if (!user.getId().equals(entity.getUser().getId())) {
+            throw new ResourceForbiddenException("Forbidden");
+        }
+
+        // update properties
+        entity.setLastUpdater(currentUser);
```
Comment (jamesrossiter) [inline root/api/src/main/java/org/patientview/api/service/impl/PathwayServiceImpl.java]: need to set last updated date

Comment (p-maks) [inline root/api/src/main/java/org/patientview/api/service/impl/PathwayServiceImpl.java]: it's on the model itself
@PreUpdate
    public void preUpdate() {
        setLastUpdate(new Date());
    }

File: root/webapp/app/scripts/app.js [hunk 4]
```diff
@@ -182,6 +182,9 @@ patientviewApp.run(['$rootScope', '$timeout', '$location', '$cookieStore', '$coo
                 if (!routeExists(RouteService.getApiDocsRoute(), data.routes)) {
                     data.routes.push(RouteService.getApiDocsRoute());
                 }
+                if (!routeExists(RouteService.getDonorPathwayRoute(), data.routes)) {
```
Comment (jamesrossiter) [inline root/webapp/app/scripts/app.js]: these are default open routes, should not be here (pulled from route table in db)

File: root/webapp/app/scripts/services/routeservice.js [hunk 5]
```diff
@@ -256,6 +256,23 @@ angular.module('patientviewApp').factory('RouteService', [function () {
                     }
                 }
             };
-        }
```
Comment (jamesrossiter) [inline root/webapp/app/scripts/services/routeservice.js]: this is pulled from routes db, shouldn't be here (unless you are just using for testing)

File: root/webapp/app/scripts/app.js (hunk 4 above)
Comment (p-maks) [inline root/webapp/app/scripts/app.js]: Added @lukefanning as it's his part

**Pull request closed without merging**